pandera/api/dataframe/model.py

```diff
@@ -271,24 +277,50 @@ def to_yaml(cls, stream: Optional[os.PathLike] = None):
         """
         return cls.to_schema().to_yaml(stream)
 
+    # Overloads specify effectively check_obj: TDataFrame -> TDataFrame[TDataFrameModel]
+    # but to do this directly would required higher kinded typevars (https://github.com/python/typing/issues/548)
+
+    @overload
+    @classmethod
+    def validate(
+        cls: Type[TDataFrameModel],
+        check_obj: pl.LazyFrame,
+        head: Optional[int] = None,
+        tail: Optional[int] = None,
+        sample: Optional[int] = None,
+        random_state: Optional[int] = None,
+        lazy: bool = False,
+        inplace: bool = False,
+    ) -> LazyFrame[TDataFrameModel]: ...
+
+    @overload
+    @classmethod
+    def validate(
+        cls: Type[TDataFrameModel],
+        check_obj: pd.DataFrame,
+        head: Optional[int] = None,
+        tail: Optional[int] = None,
+        sample: Optional[int] = None,
+        random_state: Optional[int] = None,
+        lazy: bool = False,
+        inplace: bool = False,
+    ) -> DataFrame[TDataFrameModel]: ...
+
     @classmethod
     @docstring_substitution(validate_doc=BaseSchema.validate.__doc__)
     def validate(
         cls: Type[TDataFrameModel],
-        check_obj: TDataFrame,
+        check_obj: pd.DataFrame | pl.LazyFrame,
```

Comment:
The `api.dataframe` module should be library-agnostic. Can we add the `pl.LazyFrame` annotation in the polars api?